src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h

```diff
@@ -2472,6 +2487,38 @@ ep_rt_utf8_string_strtok (
 	format, ...) \
 sprintf_s (reinterpret_cast<char *>(str), static_cast<size_t>(str_len), reinterpret_cast<const char *>(format), __VA_ARGS__)
 
+static
+inline
+bool
+ep_rt_utf8_string_replace (
+	ep_char8_t **str,
+	const ep_char8_t *strSearch,
+	const ep_char8_t *strReplacement
+)
+{
+	STATIC_CONTRACT_NOTHROW;
+	if ((*str) == NULL)
+		return false;
+
+	ep_char8_t* strFound = strstr(*str, strSearch);
+	if (strFound != NULL)
+	{
+		size_t strSearchLen = strlen(strSearch);
+		size_t newStrSize = strlen(*str) + strlen(strReplacement) - strSearchLen + 1; 
+		ep_char8_t *newStr =  reinterpret_cast<ep_char8_t *>(malloc(newStrSize));
+		if (newStr == NULL)
+		{
+			*str = NULL;
+			return false;
+		}
+		ep_rt_utf8_string_snprintf(newStr, newStrSize, "%.*s%s%s", (int)(strFound - (*str)), *str, strReplacement, strFound + strSearchLen);
```

Comment:
I would probably have done this using memcpy, its faster (lowered to intrinsic by most compilers) and no risk of having any interpretation done regarding code points (since it's UTF8) in underlying implementation of snprintf.

Reply:
True, memcpy would be a fair bit faster, but this should compile down to a bit less code as all of these strings have variable length, and in addition this is a rare path that is already quite slow such that the perf difference of using printf shouldn't be significant. I don't plan to change this.

Reply:
OK, was thinking that we added a generic replace function and that it might get called from other places in future that is not in a rare/slow code path, but guess we could optimize this later when we hit such case (and it turns out to be a problem). Just tried a very small benchmark using memcpy instead of snprintf for the above function and memcpy version was ~2.5x faster on msvc (for a ~70 byte string replacing {pid} with 123456789) timed over 1 million iterations.